tests/core/test_broker.py

```diff
@@ -301,3 +303,26 @@ def test_warning_if_unsupported_ssl_env_variables(monkeypatch: MonkeyPatch):
 
     with pytest.warns(UserWarning):
         pika._create_rabbitmq_ssl_options()
+
+
+async def test_pika_connection_error(monkeypatch: MonkeyPatch):
+    # patch PikaEventBroker to raise an AMQP connection error
+    async def connect(self) -> None:
+        raise aio_pika.exceptions.ProbableAuthenticationError("Oups")
+
+    monkeypatch.setattr(PikaEventBroker, "connect", connect)
+    cfg = read_endpoint_config(
```

Comment:
optional: I personally find it easier if all test data is directly in the test. Especially if it isn't re-used. This makes it easier to read the test as I don't have to jump to the file and also avoids us having thousands of files with often only minor differences.